File: 933.number-of-recent-calls.py
#
# @lc app=leetcode id=933 lang=python3
#
# [933] Number of Recent Calls
#

# @lc code=start
class RecentCounter:

    #Each test case will call ping with strictly increasing values of t
    #1 <= t <= 109

    def __init__(self):
        self.queue = []
        
    #The strictly increasing values of t means that we can use a queue to sovle the problem
    def ping(self, t: int) -> int:
        self.queue.append(t)
        while self.queue[0] < t - 3000:
            self.queue.pop(0) 
        return len(self.queue)
    
        


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)
# @lc code=end

File: 933.number-of-recent-calls.java
/*
 * @lc app=leetcode id=933 lang=java
 *
 * [933] Number of Recent Calls
 */
import java.util.Queue;
import java.util.LinkedList;
// @lc code=start
class RecentCounter {
    private Queue<Integer> queue;
    public RecentCounter() {
        queue = new LinkedList<>();
        
    }
    
    public int ping(int t) {
        queue.offer(t);
        int queueTop = queue.peek();
        while(queueTop < t - 3000){
            queue.poll();
            queueTop = queue.peek();
        } 
        return queue.size();
        
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */
// @lc code=end

